src/common/commonutils/FileUtils.c

```diff
@@ -448,463 +538,172 @@ int SetDirectoryAccess(const char* directoryName, unsigned int desiredOwnerId, u
-        return EINVAL;
-    }
-        
-    if (NULL != (targetHandle = setmntent(target, "w")))
+    if (FileExists(fileName) && CharacterFoundInFile(fileName, '+'))
     {
-        if (NULL != (sourceHandle = setmntent(source, "r")))
-        {
-            while (NULL != (mountStruct = getmntent(sourceHandle)))
-            {
-                if (0 != addmntent(targetHandle, mountStruct))
-                {
-                    if (0 == (status = errno))
-                    {
-                        status = ENOENT;
-                    }
-                    
-                    OsConfigLogError(log, "CopyMountFile ('%s' to '%s'): failed adding '%s %s %s %s %d %d', addmntent() failed with %d", source, target, 
-                        mountStruct->mnt_fsname, mountStruct->mnt_dir, mountStruct->mnt_type, mountStruct->mnt_opts, mountStruct->mnt_freq, mountStruct->mnt_passno, status);
-                    break;
-                }
-            }
-        }
-        else
-        {
-            if (0 == (status = errno))
-            {
-                status = ENOENT;
-            }
-            
-            OsConfigLogError(log, "CopyMountFile: could not open source file '%s', setmntent() failed (%d)", source, status);
-        }
-        
-        fflush(targetHandle);
-        endmntent(targetHandle);
-        endmntent(sourceHandle);
+        OsConfigLogError(log, "CheckNoLegacyPlusEntriesInFile(%s): there are '+' lines in file '%s'", fileName, fileName);
+        OsConfigCaptureReason(reason, "There are '+' lines in file '%s'", fileName);
+        status = ENOENT;
     }
     else
     {
-        if (0 == (status = errno))
-        {
-            status = ENOENT;
-        }
-
-        OsConfigLogError(log, "CopyMountFile: could not open target file '%s', setmntent() failed (%d)", target, status);
+        OsConfigLogInfo(log, "CheckNoLegacyPlusEntriesInFile(%s): there are no '+' lines in file '%s'", fileName, fileName);
+        OsConfigCaptureSuccessReason(reason, "There are no '+' lines in file '%s'", fileName);
     }
-        
+
     return status;
 }
 
-int SetFileSystemMountingOption(const char* mountDirectory, const char* mountType, const char* desiredOption, void* log)
+int ReplaceMarkedLinesInFile(const char* fileName, const char* marker, const char* newline, char commentCharacter, void* log)
 {
-    const char* fsMountTable = "/etc/fstab";
-    const char* mountTable = "/etc/mtab";
-    const char tempFileNameTemplate[] = "/tmp/~xtab%d";
-    const char* newLineAsIsTemplate = "\n%s %s %s %s %d %d";
-    const char* newLineAddNewTemplate = "\n%s %s %s %s,%s %d %d";
-
-    FILE* fsMountHandle = NULL;
-    FILE* mountHandle = NULL;
-    char* newLine = NULL;
-    char* tempFileNameOne = NULL;
-    char* tempFileNameTwo = NULL;
-    struct mntent* mountStruct = NULL;
-    bool matchFound = false;
-    int lineNumber = 0;
+    const char* tempFileNameTemplate = "/tmp/~OSConfig.ReplacingLines%u";

```

Comment:
Can we leverage `mktemp` here? This would make it more portable is `/tmp` isn't always the right place for temporary files. 
See https://linux.die.net/man/1/mktemp